docs/change-detection.md

```diff
@@ -0,0 +1,25 @@
+# Change Detection
+
+When changing your infrastructure (made up of a set of stacks) it's common to
+make several changes to several stacks. But now that you have multiple terraform
+states (per stack), how to apply the changes only to the affected resources?
```

Comment:
```suggestion
When working with Infrastructure as Code it's considered to be a best practice to split up and organize your IaC into several smaller and isolated stacks. Typically, each stack comes with its own Terraform state which allows us to plan and apply each stack on its own. Thus, introducing change detection enables us to only plan and apply stacks that have changed and ignoring all others which can lead to significant faster execution time.
```

Reply:
just a suggestion :

Reply:
It seems like it overlaps a little with why stacks indeed.

Reply:
gotcha - feel free to adjust however you'd like to see this :)

Reply:
I think we could use this as an introduction in the stacks part?